crates/hyperswitch_domain_models/src/router_response_types.rs

```diff
@@ -511,3 +512,73 @@ pub struct CompleteAuthorizeRedirectResponse {
     pub params: Option<masking::Secret<String>>,
     pub payload: Option<common_utils::pii::SecretSerdeValue>,
 }
+
+/// Represents details of a payment method.
+#[derive(Debug, Clone)]
+pub struct PaymentMethodDetails {
+    /// Indicates whether mandates are supported by this payment method.
+    pub supports_mandate: bool,
+    /// Indicates whether refund is supported by this payment method.
+    pub supports_refund: bool,
+    /// List of supported capture methods
+    pub supported_capture_methods: Vec<CaptureMethod>,
+}
+
+/// list of payment method types and metadata related to them
+pub type PaymentMethodTypeMetadata = HashMap<PaymentMethodType, PaymentMethodDetails>;
+
+/// list of payment methods, payment method types and metadata related to them
+pub type SupportedPaymentMethods = HashMap<PaymentMethod, PaymentMethodTypeMetadata>;
+
+#[derive(Debug, Clone)]
+pub struct ConnectorInfo {
+    /// Description of the connector.
+    pub description: String,
+
+    /// Connector Type
+    pub connector_type: PaymentConnectorCategory,
+}
+
+pub trait SupportedPaymentMethodsExt {
+    fn add(
+        &mut self,
+        payment_method: PaymentMethod,
+        payment_method_type: PaymentMethodType,
+        supports_mandate: bool,
+        supports_refund: bool,
+        supported_capture_methods: Vec<CaptureMethod>,
+    );
+}
+
+impl SupportedPaymentMethodsExt for SupportedPaymentMethods {
+    fn add(
+        &mut self,
+        payment_method: PaymentMethod,
+        payment_method_type: PaymentMethodType,
+        supports_mandate: bool,
```

Comment:
create a struct and take the struct as input, the problem with taking bool and other primitive types is that the values do not indicate the meaning and can be easily interchanged